File: Excel/662 Challenge.py
import pandas as pd
import numpy as np

path = "662 Create a Pyramid.xlsx"
test6 = pd.read_excel(path,  usecols="C:N", skiprows=1, nrows=6, header=None).fillna(" ").values

def draw_pyramid(stores):
    M = np.full((stores, stores * 2), " ", dtype=str)
    for i in range(1, stores + 1):
        row = [" "] * (stores - i) + ["/"] + ["/\\"] * (i - 1) + ["\\"] + [" "] * (stores - i)
        M[i - 1, :] = list("".join(row))
    return M

a = draw_pyramid(6)

print(np.array_equal(a, test6))
# True

File: Excel/600-699/662/662 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/662 Create a Pyramid.xlsx"
test6  = read_excel(path, range = "C2:N7", col_names = F) %>% as.matrix() %>% replace(., is.na(.), " ")

draw_pyramid <- function(stores) {
  M <- matrix(" ", nrow = stores, ncol = stores * 2)
  for (i in 1:stores) {
    M[i, ] <- strsplit(paste(c(rep(" ", stores - i), "/", rep("/\\", i - 1), "\\", rep(" ", stores - i)), collapse = ""), "")[[1]]
  }
  return(M)
}

a = draw_pyramid(6)

all(a == test6)
#> [1] TRUE